### ISSUE TITLE
file emoji/0042.png is missing

### ISSUE THREAD

**Issue by uraich** [kisvegabor/lvgl#3637]
### Perform all steps below and tick them with [x]
- [x] Read the [FAQ](https://docs.lvgl.io/master/intro/index.html#faq)
- [x] Check the related part of the [Documentation](https://docs.lvgl.io/)
- [x] Update lvgl to the latest version
- [x] Reproduce the issue in a [Simulator](https://docs.lvgl.io/latest/en/html/get-started/pc-simulator.html)

### Describe the bug
lv_example_imgfont_1.c is not working correctly because emoji files are not found

### To Reproduce
Trying to run lv_example_imgfont in the simulator

### Screenshots or video

![image](https://user-images.githubusercontent.com/1673533/187238681-22693e8e-b886-4034-8ac2-c77b9e5462dc.png)

**Comment by uraich** [comment]
I traced it down a little further:
First, it does not find the image path: A:lvgl/examples/assets/emoji"
When I replace it with ./lvgl/examples/assets/emoji it will find the F600 emoji
However, it interprets the following letters A and B also as unicode and tries to find 0041.png and 0042.png

**Comment by kisvegabor** [comment]
It works well for me if these are enabled in `lv_conf.h`:
```c
#define LV_USE_FS_STDIO 1
#if LV_USE_FS_STDIO
    #define LV_FS_STDIO_LETTER 'A'     /*Set an upper cased letter on which the drive will accessible (e.g. 'A')*/
    #define LV_FS_STDIO_PATH ""         /*Set the working directory. File/directory paths will be appended to it.*/
    #define LV_FS_STDIO_CACHE_SIZE 0    /*>0 to cache this number of bytes in lv_fs_read()*/
#endif
```

That is if there is an `'A'` driver.

**Comment by uraich** [comment]
This is what I get on my Ubuntu-22.04 system (latest updates)
![image](https://user-images.githubusercontent.com/1673533/187910811-8af13d0b-d6e0-4915-bcc1-e55795a88ab2.png)!
It tries to find 0031.png and 0032.png which corresponds to the "1" and "2" written before the first unicode. It also does not find F600.png 
If I replace A:lvgl/examples/assets/emoji by ../lvgl/examples/assets/emoji then it looks ok. However, I still get the error messages that 0031.png, 0032.png, 0041.png and 0042.png are not found.

![image](https://user-images.githubusercontent.com/1673533/187912201-a9ea47ff-3643-430b-91dc-1d566e29a610.png)

![Screenshot from 2022-09-01 14-00-44](https://user-images.githubusercontent.com/1673533/187909091-fbdb9f6f-ee6f-4e05-9b9f-4c41cccb287d.png)

The settings in lv_conf.h are as you describe. This is the standard example from the simulator with no modifications at all.
What is an "A" driver?

**Comment by kisvegabor** [comment]
> What is an "A" driver?

You an attache file system interfaces to LVGL and name them by a letter. E.g. the above [shown snippet](https://github.com/lvgl/lvgl/issues/3637#issuecomment-1234155978) attaches the stdio file system API to letter 'A'. See more [here](https://docs.lvgl.io/master/overview/file-system.html).

I don't know why the png is not found. What is the output of this?
```c
  char buf[200];
  getcwd(buf, 200);
  printf("%s\n", buf);
```

For me it's the project's root folder, where the lvgl folder is located. 

The warnings are kind of normal. Imgfont example checks if there is an image for each unicode latter. If so it uses it, if not lets the normal font to be used. The warnings could be eliminated by adding `    if(unicode < 0xF000) return false;` to `get_imgfont_path()` in the example.

**Comment by uraich** [comment]
Ok, that works. Thanks.
However, lv_imgfont_create does not adhere to the MicroPython conventions and translating 
lv_imgfont_create to imgfont=lv.imgfont(lv..scr_act()) does not work. Is there a way to correct this?

**Comment by uraich** [comment]
Hi Gabor,
In the simulator the directory in which the binary is stored is simulator/build/bin/main. When I run the program with:
./build/bin/main, while removing the drive letter, then it works. Putting the drive letter in, it fails.
How does the drive letter translate?
In the meantime I uploaded the MP version of the program onto my draft PR. There we get the directory at which the script is running and then create the png file name relative to that directory.

**Comment by kisvegabor** [comment]
The driver letter is used only to search the "driver". E.g. if you add a MicroPython FS driver with letter 'M', 'M' will be used to find the open callback of your driver. Without driver letter the open function should [return error](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_fs.c#L69).

Are you sure you are using LVGL's FS interface?

**Comment by uraich** [comment]
Hi Gabor,
In fact I am not referring to the MP example but to the C example of imgfont. When I run the program in the simulator on my Ubuntu  system this is what I see:
![image](https://user-images.githubusercontent.com/1673533/191213055-01b4f4f6-aebe-47a0-90fc-4dd3059c201f.png)
If I remove the drive letter, the example works. Did you check the program on a Unix system?

This is the entry in lv_conf.h:
/*API for fopen, fread, etc*/
#define LV_USE_FS_STDIO 1
#if LV_USE_FS_STDIO
    #define LV_FS_STDIO_LETTER 'A'      /*Set an upper cased letter on which the drive will accessible (e.g. 'A')*/
    #define LV_FS_STDIO_PATH ""         /*Set the working directory. File/directory paths will be appended to it.*/
    #define LV_FS_STDIO_CACHE_SIZE  0   /*>0 to cache this number of bytes in lv_fs_read()*/
#endif

**Comment by kisvegabor** [comment]
On Linux Mint I'm using the Eclipse project and the example works for me out of the box. 

However if I build with Cmake 
- it doesn't work
```
cd bin
./main
```
- but it works: 
```
./bin/main
```

So it depends on from where you start the application. `"A:lvgl/examples/assets/emoji"` assumes that it runs from the parent folder of `lvgl`.

**Comment by uraich** [comment]
Frankly, I don't understand this, I am using the Eclipse project as you do. I am building with cmake as you do and I am running the program in the build folder with ./bin/main as you do. But in my case it does not find the file.
I also tried to run it one level below:
./build/bin/main
from this directory I find the file with 
ls lvgl/examples/assets/emoji
but the problem is still the same.

**Comment by kisvegabor** [comment]
My directory structure looks like this:
```
├── bin
├── build
├── CMakeLists.txt
├── Debug
├── Dockerfile
├── licence.txt
├── lv_conf.h
├── lv_drivers
├── lv_drv_conf.h
├── lvgl
├── main.c
├── Makefile
├── mouse_cursor_icon.c
├── pc_simulator.launch
└── README.md
```

So `bin` and `build` are next to each other, not `bin` inside `build`.

**Comment by uraich** [comment]
I tried to figure out where the problem comes from and I found that the filename is passed as A:lvgl/examples/assets/emoji/F600.png
to avformat_open_input in ffmpeg_get_img_header
I think that at this moment the drive name should be gone.
Do you have ffmpeg enabled in your system?
![image](https://user-images.githubusercontent.com/1673533/191283684-fc5da5fe-9adf-4b35-9013-9cf52e479392.png)

**Comment by kisvegabor** [comment]
We have modified all the image decoders (PNG, GIF, JPG, etc) except ffmpeg to use LVGL files system (the others are mainly 1 file libs, so it was easy to refactor). 

I know, it's a little bit messy this way, but I can't see a uniform solution. (Maybe having a default FS driver if there is no letter?) So what happens is that "A:path/file" doesn't work for ffmepeg because it really  interprets the path as it is (with "A:"). So without driver letter somehow ffmpeg finds the image. But I don't know why the same path with driver letter doesn't work for other decoders. 

In other words the same path relative to the current working directory (whatever it is) should work in both cases or none of of them as the only difference is that in one case an `open` function in ffmpeg tries to open it, in the other case an `open` function from PNG decoder through LVGL's FS interface tries to open it. 

To debug it you can add a a `printf` in `lv_fs_stdio.c`'s `open` function to see the path it gets.

**Comment by uraich** [comment]
If I disable ffmpeg and I enable the PNG decoder library, then everything works.
Should we check in the example code if FFMPEG is enabled in which case we remove the drive letter? Then it should work in both cases.

**Comment by kisvegabor** [comment]
For me it works as I expected. 

I added this line at the beginning of `get_imgfont_path` to get rid of some irrelevant log messages:
```c
    if(unicode < 0xF000) return false;
```

After that with
```c
"lvgl/examples/assets/emoji"
```
the images are displayed and I get these logs:
```
[Warn]	(0.121, +2)	 lv_fs_open: Can't open file (lvgl/examples/assets/emoji/F600.png): unknown driver letter 	(in lv_fs.c line #68)
```

It makes sense, as the PNG and other decoders couldn't interpret the path.

With
```c
"A:lvgl/examples/assets/emoji"
```
I get not logs as the PNG decoder could open the image without any issues.

I tried all these from Eclipse without Cmake.

**Comment by uraich** [comment]
When I do not enable FFMPEG and I remove the drive letter, then I get the above warning and the image does **not** show
I tried this:
#if LV_USE_FFMPEG
	snprintf(path, len, "%s/%04X.%s", "lvgl/examples/assets/emoji", unicode, "png");
#else
	snprintf(path, len, "%s/%04X.%s", "A:lvgl/examples/assets/emoji", unicode, "png");
#endif
and this works in all cases fro me

**Comment by kisvegabor** [comment]
> When I do not enable FFMPEG and I remove the drive letter, then I get the above warning and the image does not show

It makes sense. Without FFMPEG there is no decoder which can open the path directly (without driver letter).

To better illustrate the actual case:
```c
#if LV_USE_FFMPEG
snprintf(path, len, "%s/%04X.%s", "lvgl/examples/assets/emoji", unicode, "png");
#elif LV_USE_PNG
snprintf(path, len, "%s/%04X.%s", "A:lvgl/examples/assets/emoji", unicode, "png");
#endif
```

So.. wait! You have `LV_USE_PNG` enabled, right?

**Comment by uraich** [comment]
Yes, LV_USE_PNG is enabled

**Comment by kisvegabor** [comment]
If for any reason [this](https://github.com/lvgl/lvgl/issues/3637#issuecomment-1252744222) works better for you, let's go with it. I updated the the example in 8f3f60337.
